docs/DeveloperGuide.md

```diff
@@ -528,7 +571,94 @@ end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline
   * Cons: Hard to implement as a new form of storage or memory has to be created to maintain the relative ordering of
     events.
 
+### Make Statistics
+
+The Make Statistics feature that is accessed through the `makeStats` command allows users to generate statistics of the persons of the `Person` class that are tagged to an event of the `Event` class, and show a piechart of the statistics in a new window.
+
+There are 2 types of statistics that can be generated:
+* Age: Shows the proportion of ages of the `persons` tagged to the `event` based on 5-year age ranges.
+* Gender: Shows the proportion of genders of the `persons` tagged to the `event`, seperating them into either `Male` or `Female`.
+
+Users specify the type by specifying the 2 fields below:
+- Index of the `event`
+- Type of statistic to generate from the `event`
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** Age is generated using the date of birth of the persons tagged to the event and is calculated from the current real-time date.
+</div>
+
+The `makeStats` operation is facilitated by `MakeStatsCommand` which extends from `Command`. If the users' input matches
+the `COMMAND_WORD` of `MakeStatsCommand` in `AddressBookParser#parseCommand()`, `MakeStatsCommandParser#parse()` will
+process the additional user input which constitutes the 6 compulsory fields and return an `AddPersonCommand`.
+
+Executing this Command object through the `MakeStatsCommand#execute()` first triggers the `Model` interface's
+`Model#setData()`. This operation updates the list of piechart data in the `Model`.
 
+ Then, the `Logic` interface's `Logic#setPieChartData()` is triggered. This operation subsequently calls upon the `MainWindow#setPieChart()` operation which in turn calls upon the `MainWindow#handleStats()` operation, which finally culminates with a call upon the `StatsWindow#show()` which displays the piechart in a new window.
+
+The following sequence diagram shows the methods calls related to the make stats operation:
+
+![AddPersonSequenceDiagram](images/MakeStatsSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** `cmd` in the diagram represents the make stats command text entered by user.
+</div>
+
+### Tagging Events
+
+The Tag Event feature that is accessed through the `tagEvent` command allows users to tag persons from existing marketing campaigns of the `event` class in the application.
+
+In order to tag persons from `events`, users need to specify:
+- Index of the `event`
+- Index(s) of the `persons` to tag from the `event`
```

Comment:
I think we shouldn't put `persons` here since `persons` is not really an object or class, but the change below also looks abit weird
```suggestion
- Index(s) of the `person`s to tag from the `event`
```

Reply:
yea... was why i put it inside but i think this is more correct so ill just put this